isis/tests/FunctionalTestsPhocube.cpp

```diff
@@ -346,4 +353,38 @@ TEST_F(DefaultCube, FunctionalTestPhocubeNoBandBin) {
   EXPECT_NEAR(hist->StandardDeviation(), 7.3598007219398722, .0001);
 
   cube.close();
-}
\ No newline at end of file
+}
+
+
+TEST_F(DefaultCube, FunctionalTestPhocubeAllDnBands) {
+  QString cubeFileName = tempDir.path() + "/phocubeTEMP.cub";
+  QVector<QString> args = {"to=" + cubeFileName, "alldn=true"};
+
+  UserInterface options(APP_XML, args);
+  resizeCube(5, 5, 3);
+  Pvl *inIsisLabel = testCube->label();
+  PvlGroup &inBandBin = inIsisLabel->findGroup("BandBin", Pvl::Traverse);
+  inBandBin["FilterName"] = "(B1, B2, B3)";
+
+  phocube(testCube, options);
+
+  Cube cube(cubeFileName);
+  Pvl *isisLabel = cube.label();
+
+  ASSERT_EQ(cube.sampleCount(), testCube->sampleCount());
+  ASSERT_EQ(cube.lineCount(), testCube->lineCount());
+  ASSERT_EQ(cube.bandCount(), 8);
+
+  PvlGroup bandBin = isisLabel->findGroup("BandBin", Pvl::Traverse);
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, bandBin.findKeyword("FilterName")[0], "B1");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, bandBin.findKeyword("FilterName")[1], "B2");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, bandBin.findKeyword("FilterName")[2], "B3");
+
+  std::unique_ptr<Histogram> hist (cube.histogram(0));
+  EXPECT_NEAR(hist->Average(), 67.575204, .000001);
+  EXPECT_NEAR(hist->Sum(), 13515.040821, .000001);
+  EXPECT_EQ(hist->ValidPixels(), 200);
+  EXPECT_NEAR(hist->StandardDeviation(), 76.089184, .000001);
```

Comment:
Instead of hard coded values, pull the histogram from the input cube and its it for comparison.

Reply:
done